spec/double_entry/line_spec.rb

```diff
@@ -31,7 +31,14 @@
 
       context 'given code = nil' do
         let(:code) { nil }
-        specify { expect { line_to_persist.save! }.to raise_error }
+        let(:expected_error) do
+          if defined?(ActiveRecord::NotNullViolation)
+            ActiveRecord::NotNullViolation
+          else
+            ActiveRecord::StatementInvalid
+          end
```

Comment:
`ActiveRecord::NotNullViolation` seems new in Rails 5.1. Prior versions raise `ActiveRecord::StatementInvalid`.